adobe_vipm/flows/fulfillment/configuration.py

```diff
@@ -0,0 +1,134 @@
+"""
+This module contains the logic to implement the configuration fulfillment flow.
+It exposes a single function that is the entrypoint for configuration order
+processing.
+"""
+import logging
+
+from adobe_vipm.adobe.client import get_adobe_client
+from adobe_vipm.adobe.errors import AdobeError, SubscriptionNotFoundError
+from adobe_vipm.flows.context import Context
+from adobe_vipm.flows.fulfillment.shared import (
+    CompleteConfigurationOrder,
+    ConfigurationStartOrderProcessing,
+    switch_order_to_failed,
+)
+from adobe_vipm.flows.helpers import SetupContext
+from adobe_vipm.flows.pipeline import Pipeline, Step
+from adobe_vipm.flows.utils import (
+    get_adobe_customer_id,
+    notify_not_updated_subscriptions,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def fulfill_configuration_order(client, order):
+    """
+    Fulfills a configuration order.
+    """
+    logger.info(f'Start processing {order["type"]} order {order["id"]}')
+
+    pipeline = Pipeline(
+        SetupContext(),
+        ConfigurationStartOrderProcessing(),
+        SubscriptionUpdateAutoRenewal(),
+        CompleteConfigurationOrder(),
+    )
+
+    context = Context(order=order)
+    pipeline.run(client, context)
+
+
+
+class SubscriptionUpdateAutoRenewal(Step):
+    def __call__(self, client, context, next_step):
+        """
+        Updates the auto renewal status of a subscription.
+        """
+        logger.info(f'Start processing {context.order["type"]} order {context.order["id"]}')
+
+        adobe_client = get_adobe_client()
+        customer_id = get_adobe_customer_id(context.order)
+        authorization_id = context.order["authorization"]["id"]
+        update_success_subscriptions = []
+
+        for subscription in context.order["subscriptions"]:
+            subscription_vendor_id = subscription["externalIds"]["vendor"]
+            desired_auto_renewal = subscription["autoRenew"]
+            desired_quantity = subscription["lines"][0]["quantity"]
+
+            adobe_subscription = adobe_client.get_subscription(
+                authorization_id,
+                customer_id,
+                subscription_vendor_id)
+
+            try:
+                if not adobe_subscription:
+                    logger.info(
+                        f"No matching Adobe subscription found for vendor_id "
+                        f"{subscription_vendor_id} in order {context.order['id']}"
+                    )
+                    raise SubscriptionNotFoundError("No matching Adobe subscription found")
```

Comment:
Looks really strange that you raise this exception and handle it in the same block... Ideally it should be either inside the `get_subscription` or you should run the function

What I propose - is to move

1. Move this block outside the try
2. create a separate function for exception handling
3. use this function both in except block and in `if`
4. Do not raise exception